napalm_nxos/nxos.py

```diff
@@ -241,6 +241,46 @@ def get_lldp_neighbors(self):
 
         return results
 
+
+    def get_bgp_neighbors(self):
+        cmd = 'show bgp sessions vrf all'
+        tbl = self._get_command_table(cmd, 'TABLE_vrf', 'ROW_vrf')
+
+        ret = {}
+        for el in tbl:
```

Comment:
Renamed.  I called things that were in the output `result_xxx_dict`, vs things that were in the input.  Too much hash juggling, too many vars.

fyi, I could have extracted a helper method to refactor this and make it clearer, but I figured we'd be hacking on it in the coming days/weeks soon, so I'm holding off on doing that until we work out what the method should actually be doing.  Then I can refactor it afterwards.  I have a few test cases in the mock data files so the coverage should be OK.